long-ride-alerts/src/main/java/org/apache/flink/training/exercises/longrides/LongRidesExercise.java

```diff
@@ -98,17 +101,60 @@ public static void main(String[] args) throws Exception {
     @VisibleForTesting
     public static class AlertFunction extends KeyedProcessFunction<Long, TaxiRide, Long> {
 
+        private ValueState<TaxiRide> rideState;
+        public static final Duration RIDE_TIMEOUT = Duration.ofHours(2);
```

Comment:
Good manner would be move
`public static final` declaration at the beginning of vars declaration
and then we should provide non-static fields like `rideState`